pkg/network/node/pod.go

```diff
@@ -67,18 +72,27 @@ type podManager struct {
 
 	// Things only accessed through the processCNIRequests() goroutine
 	// and thus can be set from Start()
-	ipamConfig []byte
+	ipamConfig   []byte
+	podInformers informers.SharedInformerFactory
+	listerSynced cache.InformerSynced
 }
 
+const defaultPodInformerResyncPeriod = 30 * time.Minute
+
 // Creates a new live podManager; used by node code0
-func newPodManager(kClient kubernetes.Interface, policy osdnPolicy, overlayMTU uint32, routableMTU uint32, ovs *ovsController) *podManager {
+func newPodManager(kClient kubernetes.Interface, policy osdnPolicy, overlayMTU uint32, routableMTU uint32, ovs *ovsController, nodeName string) *podManager {
 	pm := newDefaultPodManager()
 	pm.kClient = kClient
 	pm.policy = policy
 	pm.overlayMTU = overlayMTU
 	pm.routableMTU = routableMTU
 	pm.podHandler = pm
 	pm.ovs = ovs
+	pm.podInformers = kinformers.NewSharedInformerFactoryWithOptions(kClient, defaultPodInformerResyncPeriod,
+		informers.WithTweakListOptions(func(options *metav1.ListOptions) {
+			options.FieldSelector = fields.OneTermEqualSelector("spec.nodeName", nodeName).String()
```

Comment:
The process already has an all-pods watcher (in networkpolicy.go) so it would be more efficient to just use the same shared informer in this file (and manually filter by node name in the callback)

Reply:
this is customized informer to just fill up the cache to speed up the reattach without registered event handler